src/test/java/org/ow2/petals/deploymentmodel/DeployTest.java

```diff
@@ -0,0 +1,239 @@
+ * option) any later version.
+ *
+ * This program/library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program/library; If not, see http://www.gnu.org/licenses/
+ * for the GNU Lesser General Public License version 2.1.
+ */
+
+package org.ow2.petals.deploymentmodel;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.ConsoleHandler;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.ow2.petals.admin.junit.ArtifactLifecycleFactoryMock;
+import org.ow2.petals.admin.junit.PetalsAdministrationApi;
+import org.ow2.petals.admin.topology.Container.PortType;
+import org.ow2.petals.admin.topology.Container.State;
+import org.ow2.petals.deploymentmodel.busmodel.Bus;
+import org.ow2.petals.deploymentmodel.busmodel.BusModel;
+import org.ow2.petals.deploymentmodel.busmodel.ComponentInstance;
+import org.ow2.petals.deploymentmodel.busmodel.ContainerInstance;
+import org.ow2.petals.deploymentmodel.busmodel.Machine;
+import org.ow2.petals.deploymentmodel.busmodel.ProvisionedMachine;
+import org.ow2.petals.deploymentmodel.busmodel.ServiceUnitInstance;
+import org.ow2.petals.deploymentmodel.busmodel.TopologyInstance;
+import org.ow2.petals.deploymentmodel.componentrepository.Component;
+import org.ow2.petals.deploymentmodel.componentrepository.ComponentRepository;
+import org.ow2.petals.deploymentmodel.serviceunitmodel.ServiceUnit;
+import org.ow2.petals.deploymentmodel.serviceunitmodel.ServiceUnitModel;
+import org.ow2.petals.deploymentmodel.topologymodel.Container;
+import org.ow2.petals.deploymentmodel.topologymodel.Topology;
+import org.ow2.petals.deploymentmodel.topologymodel.TopologyModel;
+import org.ow2.petals.jbi.descriptor.original.JBIDescriptorBuilder;
+
+import com.google.common.io.Resources;
+
+public class DeployTest {
+
+    final public static String CONTAINER_NAME = "sample-0";
+
+    final public static String CONTAINER_HOST = "localhost";
+
+    final public static int CONTAINER_JMX_PORT = 7700;
+
+    final public static String CONTAINER_USER = "petals";
+
+    final public static String CONTAINER_PWD = "petals";
+
+    final public static State CONTAINER_STATE = State.REACHABLE;
+
+    @Rule
+    public PetalsAdministrationApi petalsAdminApiRule = new PetalsAdministrationApi();
+
+    @Test
+    public void demoSoap() throws Exception {
+        Logger logger = Logger.getLogger(ModelDeployer.class.getName());
```

Comment:
Use a conf file available in test resources that you load through the 1st method annotated `@BeforeClass`:
```
        final InputStream isLogCfg = Thread.currentThread().getContextClassLoader()
                .getResourceAsStream("org/ow2/petals/component/framework/process/mexProcessorlog.properties");
        assertNotNull(isLogCfg);
        LogManager.getLogManager().readConfiguration(isLogCfg);
```

Reply:
I don't use a config file, but I moved this part in a `@BeforeClass` method.